org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/date/StrictSimpleDateFormat.java

```diff
@@ -6,10 +6,21 @@
 import java.util.Date;
 import java.util.Locale;
 
+import org.eclipse.scout.rt.platform.util.StringUtility;
+
 /**
  * Strict variant of {@link SimpleDateFormat}. When parsing a string, it only returns a valid date when the input string
  * matches the pattern <b>exactly</b>.
  * <p>
+ * This prevents the parsing of strings with date and time information when the pattern only contains the date part but
+ * no time zone information. The default {@link SimpleDateFormat} implementation silently consumes the "surplus" time
+ * information, even when the lenient flag is disabled. This could lead to a "shifted day" problem when the resulting
+ * date object is formatted again with the same pattern. Example: {@code "1970-01-01 00:00"} → (parse) → <i>1969-12-31
+ * 22:00</i> → (format) → {@code "1969-12-31"}.
+ * <p>
+ * <b>Note:</b> When the pattern string contains a time zone character ({@code "Z"}, {@code "z"} or {@code "X"}), no
```

Comment:
Move this note after the section about "The behavior can be controlled by setting the {@link #setLenient(boolean)}" since the current order suggests that the special handling for time zone characters may be influenced by turing on lenient mode.